### PR TITLE
chore: add conformance suite client options

### LINKED ISSUE #3342: Conformance tests cannot run on v1.2.0
**What happened**: Almost all conformance tests fail on 1.2.0 with errors like `no kind is registered for the type v1.Gateway in scheme`

**What you expected to happen**: Conformance tests can run

This is caused by https://github.com/kubernetes-sigs/gateway-api/pull/3211. This overrides the passed in `Client`, `Clientset` and `RestConfig`. This overrides any customizations we make to those, notably the scheme (but not ONLY the scheme, so its likely we will want a fix that is not scoped to that).

I believe how its working in any case is due to controller-runtime using the global scheme from client-go by default, then `DefaultOptions` taking that and installing. If you don't use `DefaultOptions` then you don't hit this path

### PR DESCRIPTION
**Pull request by BobyMCbobs**
**What type of PR is this?**

/kind test
/kind bug
/kind cleanup
/area conformance

**What this PR does / why we need it**:

pass options to the Kubernetes client to allow for things like schema and mapper.

**Which issue(s) this PR fixes**:

Fixes #3342

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: conformance/utils/suite/suite.go [hunk 1]
```diff
@@ -234,6 +236,7 @@ func NewConformanceTestSuite(options ConformanceOptions) (*ConformanceTestSuite,
 
 	suite := &ConformanceTestSuite{
 		Client:           options.Client,
```
Comment (howardjohn) [inline conformance/utils/suite/suite.go:238]: Is there much point allowing someone to pass in a `Client` anymore if its going to be ignored anyways?

**Review by howardjohn — COMMENTED** [review]
LGTM, this should fix our issue I think. Appreciate the quick turnaround

File: conformance/utils/suite/suite.go (hunk 1 above)
Comment (BobyMCbobs) [reply conformance/utils/suite/suite.go:238]: This is a fair comment. It should be removed

Comment (BobyMCbobs) [reply conformance/utils/suite/suite.go:238]: Would a follow up PR be suitable?

**Review by howardjohn — APPROVED** [review]

**Pull request merged**